packages/discovery-provider/src/tasks/index_payment_router.py

```diff
@@ -0,0 +1,858 @@
+    else None
+)
+
+# The amount of USDC that represents one USD cent
+USDC_PER_USD_CENT = 10000
+
+
+def refresh_user_balances(session: Session, redis: Redis, accts=List[str]):
+    results = (
+        session.query(User.user_id, UserBankAccount.bank_account)
+        .join(
+            UserBankAccount,
+            and_(
+                UserBankAccount.bank_account.in_(accts),
+                UserBankAccount.ethereum_address == User.wallet,
+            ),
+        )
+        .filter(User.is_current == True)
+        .all()
+    )
+    # Only refresh if this is a known account within audius
+    if results:
+        user_ids = [user_id[0] for user_id in results]
+        logger.info(f"index_payment_router.py | Refresh user_ids = {user_ids}")
+        enqueue_immediate_balance_refresh(redis, user_ids)
+    return results
+
+
+def get_sol_tx_info(
+    solana_client_manager: SolanaClientManager,
+    tx_sig: str,
+):
+    try:
+        tx_info = solana_client_manager.get_sol_tx_info(tx_sig)
+        return (tx_info, tx_sig)
+    except SolanaTransactionFetchError:
+        return None
+
+
+def get_track_owner_id(session: Session, track_id: int) -> Optional[int]:
+    """Gets the owner of a track"""
+    track_owner_id = (
+        session.query(Track.owner_id).filter(Track.track_id == track_id).first()
+    )
+    if track_owner_id is not None:
+        return track_owner_id[0]
+    else:
+        return None
+
+
+# Used to limit tx history if needed
+MIN_SLOT = int(shared_config["solana"]["payment_router_min_slot"])
+INITIAL_FETCH_SIZE = 10
+
+# Used to find the correct accounts for sender/receiver in the transaction and the PaymentRouterPDA
+TRANSFER_INSTRUCTION_SENDER_ACCOUNT_INDEX = 0
+TRANSFER_INSTRUCTION_MINT_ACCOUNT_INDEX = 1
+TRANSFER_INSTRUCTION_RECEIVER_ACCOUNT_INDEX = 2  # Should be the PaymentRouterPDA
+
+ROUTE_INSTRUCTION_SENDER_INDEX = 0
+# This should be an account owned by the PaymentRouterPDA
+ROUTE_INSTRUCTION_PAYMENT_ROUTER_PDA_INDEX = 1
+ROUTE_INSTRUCTION_SPL_TOKEN_PROGRAM_INDEX = 2
+# Everything after this index is a receiver account
+ROUTE_INSTRUCTION_RECEIVER_ACCOUNTS_START_INDEX = 3
+
+
+def check_config():
+    if not all([WAUDIO_MINT_PUBKEY, USDC_MINT_PUBKEY, PAYMENT_ROUTER_PUBKEY]):
+        logger.error(
+            f"index_payment_router.py | Missing required configuration"
+            f"WAUDIO_MINT_PUBKEY: {WAUDIO_MINT_PUBKEY} USDC_MINT_PUBKEY: {USDC_MINT_PUBKEY} PAYMENT_ROUTER_PUBKEY: {PAYMENT_ROUTER_PUBKEY}- exiting."
+        )
+        return False
+    return True
+
+
+class ReceiverUserAccount(TypedDict):
+    user_id: int
+    user_bank_account: str
+
+
+class PurchaseMetadataDict(TypedDict):
+    price: int
+    splits: dict[str, int]
+    type: PurchaseType
+    id: int
+    purchaser_user_id: int
+    content_owner_id: int
+
+
+# Return highest payment router slot that has been processed
+def get_highest_payment_router_tx_slot(session: Session):
```

Comment:
nit and you don't need to do it now, but i think we should prefer using the system that `index_spl_token.py` follows where it stores a single row containing the latest transaction.
ideally we update user bank and reward manager indexers to do this too.